osu.Game/OsuGameBase.cs

```diff
@@ -82,35 +83,41 @@ public OsuGameBase()
         protected override IReadOnlyDependencyContainer CreateLocalDependencies(IReadOnlyDependencyContainer parent) =>
             dependencies = new DependencyContainer(base.CreateLocalDependencies(parent));
 
-        private SQLiteConnection createConnection()
+        private OsuDbContext createDbContext()
         {
-            var conn = Host.Storage.GetDatabase(@"client");
-            conn.BusyTimeout = new TimeSpan(TimeSpan.TicksPerSecond * 10);
-            return conn;
+            var connectionString = Host.Storage.GetDatabaseConnectionString(@"client");
+            var context = new OsuDbContext(connectionString);

```

Comment:
this is all stupid. why is this not at the framework level? why is sqlite implemented in osu.game? why is there even anything left in the framework? it's pointless at this point because all the implementation (except for the connection string?) is here.

and it's in OsuGameBase

no please god no

Reply:
I can't create OsuDbContext at the framework level because it must contain DbSets(which are game specific).

Reply:
if t can’t be abstracted to the framework then it shouldn’t exist in the framework, but i’m pretty sure it can be.